experimental/displaynames/tests/tests.rs

```diff
@@ -47,6 +47,10 @@ fn test_concatenate() {
             input_1: &Locale::from_str("es-419-fonipa").unwrap(),
             expected: "Latin American Spanish (IPA Phonetics)",
         },
+        TestCase {
+            input_1: &Locale::from_str("es-Latn-419").unwrap(),
+            expected: "Spanish (Latin, Latin America)",
```

Comment:
Shouldn't this be `Latin American Spanish (Latin)` ?

Reply:
But "es-Latn" doesn't exist in locale data? Don't we consider just the sequential subtags to get the longest matching string for locale display name? For example the test for "es-Cyrl-MX" returns "Spanish (Cyrillic, Mexico)" and not the "Mexican Spanish (Cyrillic)".

Reply:
hmm, I think you're right: the CLDR test data does the same thing

https://github.com/unicode-org/cldr/blob/main/common/testData/localeIdentifiers/localeDisplayName.txt#L15

this is different than my intuitive sense but if it's what's in the spec then that's what we should implement!

Reply:
Locale data contains
```json
"es-419": "Latin American Spanish",
```

so the result should be "Latin American Spanish (Latin)". The condition that stops this from happening is in line 463: LR is only tried if the locales has no script. I don't think that's following the spec, because `es-419`, not `es` should be the longest matching subtag 

> Match the L subtags against the type values in the <language> elements. Pick the element with the most subtags matching.

Reply:
I do think non-sequential subtags can be a valid match, because the spec also says 

> If there is more than one such element, pick the one that has subtypes matching earlier.

This kind of implies that there can be multiple matches that aren't prefixes of one another, so subtags can be skipped.

Reply:
Okay, I think the inconsistency with the CLDR test data is that it is running in non-dialect mode. In dialect mode, it's pretty clear that "Latin American Spanish (Latin)" is the correct output. This can be seen by the fact that the CLDR test data for es-419 also says "Spanish (Latin America)" instead of "Latin American Spanish".

So, yes, please fix this.

Reply:
(happy doing this in a follow-up in #3913)